addons/hr_work_entry_holidays/models/hr_version.py

```diff
@@ -66,3 +69,151 @@ def _get_sub_leave_domain(self):
+
+        employees = self.mapped('employee_id')
+        resources = employees.mapped('resource_id')
+        calendars = self.mapped('resource_calendar_id')
+
+        leaves_by_employee = dict(self.env['hr.leave']._read_group(
+            domain=[
+                ('employee_id', 'in', employees.ids),
+                ('state', '=', 'validate'),
+                ('date_from', '<=', end_dt.replace(tzinfo=None)),
+                ('date_to', '>=', start_dt.replace(tzinfo=None)),
+            ],
+            groupby=['employee_id'],
+            aggregates=['id:recordset'],
+        ))
+
+        if not leaves_by_employee:
+            return work_entry_values
+
+        existing_entries = {(vals['date_start'], vals['date_stop']) for vals in work_entry_values}
+        public_holidays = self.env['resource.calendar.leaves'].search([
+            ('resource_id', '=', False),
+            ('company_id', 'in', self.company_id.ids),
+            ('calendar_id', 'in', calendars.ids + [False]),
+            ('date_from', '<=', end_dt.replace(tzinfo=None)),
+            ('date_to', '>=', start_dt.replace(tzinfo=None)),
+        ])
+
+        attendance_intervals_by_calendar = {
+            calendar: calendar._attendance_intervals_batch(start_dt, end_dt, resources=resources)
+            for calendar in calendars
+        }
+
+        real_attendance_by_leave = {}
+        for employee, leaves in leaves_by_employee.items():
+            for leave in leaves:
+                attendances = leave.resource_calendar_id._get_actual_attendances(
+                    max(localized(leave.date_from), start_dt),
+                    min(localized(leave.date_to), end_dt)
+                )
+                real_attendance_by_leave.setdefault(employee, []).extend(attendances)
+
+        for version in self:
+            employee = version.employee_id
+            resource = employee.resource_id
+            calendar = version.resource_calendar_id or version.company_id.resource_calendar_id
+            tz = pytz.timezone(employee.tz)
+
+            attendance_intervals = attendance_intervals_by_calendar.get(calendar, {}).get(resource.id, [])
+            attendance_dates = {start.date() for start, *_ in attendance_intervals}
+
+            real_attendance = real_attendance_by_leave.get(employee, [])
+            if real_attendance:
+                # Generate normal attendance work entries for the remaining portion of the day
+                # when an employee applies for a half-day or custom-hours leave.
+                real_attendance_intervals = Intervals([
+                    (tz.localize(start), tz.localize(end), attendance)
+                    for start, end, attendance in real_attendance
+                ])
+                default_work_entry_type = self._get_default_work_entry_type_id()
+                for interval_start_dt, interval_end_dt, _attendance in real_attendance_intervals - attendance_intervals:
```

Comment:
What's different between `real_attendance_intervals` and `attendance_intervals` besides the absence of a timezone in the first one?